pyqs/worker.py

```diff
@@ -267,6 +267,164 @@ def _run_hooks(self, hook_name, context):
             hook(context)
 
 
+class SimpleProcessWorker(BaseWorker):
+
+    def __init__(self, queue_url, batchsize,
+                 connection_args=None, *args, **kwargs):
+        super(SimpleProcessWorker, self).__init__(*args, **kwargs)
+        if connection_args is None:
+            connection_args = {}
+        self.connection_args = connection_args
+        self.queue_url = queue_url
+
+        sqs_queue = get_conn(**self.connection_args).get_queue_attributes(
+            QueueUrl=queue_url, AttributeNames=['All'])['Attributes']
+        self.visibility_timeout = int(sqs_queue['VisibilityTimeout'])
+
+        self.batchsize = batchsize
+        self._messages_to_process_before_shutdown = 100
+        self.messages_processed = 0
+
+    def run(self):
+        # Set the child process to not receive any keyboard interrupts
+        signal.signal(signal.SIGINT, signal.SIG_IGN)
+
+        logger.info(
+            "Running SimpleProcessWorker: {}, pid: {}".format(
+                self.queue_url, os.getpid()))
+
+        while not self.should_exit.is_set() and self.parent_is_alive():
+            self.read_message()
+
+            if self.messages_processed \
+                    >= self._messages_to_process_before_shutdown:
+                self.shutdown()
```

Comment:
You asked about whether or not we want to keep `messages_processed`. What are the shutdown implications if we don't? I'm not worried about the memory from keeping a counter.

Reply:
In hindsight, it's probably safer to keep, so unless we all strongly disagree, we will most likely just keep it, but I wanted it to surface.